File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/deepseek-coder-33b-instruct/Avatar/Python/Java/codeforces_242_B.java
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int intersection_id = 0;
        int l = scanner.nextInt();
        int r = scanner.nextInt();
        int check_condition_1 = 837;
        int check_condition_2 = 707;
        int outer_loop_start = 919;
        int outer_loop_end = 918;
        for (int LoopIndexOut = outer_loop_start / outer_loop_end; LoopIndexOut > 0; LoopIndexOut--) {
            recursive_check(1, n, 1, scanner, l, r, intersection_id, check_condition_1, check_condition_2);
        }
        System.out.println(intersection_id == n ? -1 : intersection_id + 1);
    }

    private static void recursive_check(int i, int stop, int step, Scanner scanner, int l, int r, int intersection_id, int check_condition_1, int check_condition_2) {
        if (step == 0 || (step > 0 && i >= stop) || (step < 0 && i <= stop)) {
            return;
        }
        int li = scanner.nextInt();
        int ri = scanner.nextInt();
        if ((check_condition_1 & check_condition_2) != 0) {
            if (li <= l && r <= ri) {
                intersection_id = i;
            } else if (li < l || r < ri) {
                intersection_id = -1;
            }
        }
        l = Math.min(l, li);
        r = Math.max(r, ri);
        recursive_check(i + step, stop, step, scanner, l, r, intersection_id, check_condition_1, check_condition_2);
    }
}
```

File: dataset/Intermediate/Translation/Avatar/v0-Python-code/codeforces_242_B.py
import numpy as np
import os, sys, io, math
from tokenize import Triple
from math import *
I = lambda : [*map(int, sys.stdin.readline().split())]
IS = lambda : input()
read_int = lambda : int(input())
read_float = lambda : float(input())
n = read_int()
intersection_id = 0
(l, r) = map(int, input().split())
check_condition_1 = 837
check_condition_2 = 707
outer_loop_start = 919
outer_loop_end = 918
for LoopIndexOut in range(outer_loop_start // outer_loop_end):

    def recursive_check(i, stop, step):
        global l, r, intersection_id
        if step == 0 or (step > 0 and i >= stop) or (step < 0 and i <= stop):
            return
        (li, ri) = map(int, input().split())
        if check_condition_1 & check_condition_2:
            if li <= l and r <= ri:
                intersection_id = i
            elif li < l or r < ri:
                intersection_id = n
        l = np.min(np.array([l, li]))
        r = np.max(np.array([r, ri]))
        recursive_check(i + step, stop, step)
    recursive_check(1, n, 1)
print(-1 if intersection_id == n else intersection_id + 1)